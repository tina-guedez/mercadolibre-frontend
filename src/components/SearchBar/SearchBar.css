.nav-search {
  grid-column: 3 / 12;
  grid-row: 1;
  height: 40px;
  margin: 1px;
  margin-left: -20px;
  position: relative;
  padding: 10px 0;
}

.nav-search-input {
  border: 0 rgba(0, 0, 0, 0.2);
  border-top-left-radius: 4px;
  border-bottom-left-radius: 4px;
  color: #333;
  font-size: 18px;
  font-family: Arial, Helvetica, sans-serif;
  height: 40px;
  line-height: 40px;
  margin: 0;
  outline: 0;
  padding: 0;
  padding-left: 15px;
  width: calc(100% - 56px);
}

.nav-search-input::placeholder {
  color: #999;
}

.nav-search-btn {
  background: 0 0;
  background-color: #eee;
  border: none;
  border-top-right-radius: 4px;
  border-bottom-right-radius: 4px;
  color: #333;
  font-size: 22px;
  height: 40px;
  line-height: 1em;
  margin: 10px 0px;
  outline: 0;
  position: absolute;
  padding: 0;
  right: 0;
  top: 0;
  width: 45px;
}
.nav-search-btn img {
  margin-top: 5px;
  width: 20px;
}
