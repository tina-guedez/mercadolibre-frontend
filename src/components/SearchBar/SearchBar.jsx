import './Searchbar.css';
import React from 'react';
import SearchIcon from '../../assets/ic_Search@2x.png.png';

export const SearchBar = () => (
  <form className="nav-search" action="/items" method="GET" role="search">
    <input
      aria-label="IngresÃ¡ lo que quieras encontrar"
      autoComplete="off"
      className="nav-search-input"
      maxLength="120"
      name="q"
      placeholder="Nunca dejes de buscar"
      required
      tabIndex="0"
      type="text"
    />
    <button type="submit" className="nav-search-btn" tabIndex="-1">
      <img aria-label="Buscar" className="nav-icon-search" src={SearchIcon} />
    </button>
  </form>
);
