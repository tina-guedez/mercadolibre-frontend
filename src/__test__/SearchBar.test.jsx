import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import SearchBar from '../components/SearchBar/SearchBar';

describe('Pruebas en el componente SearchBar', () => {
  // eslint-disable-next-line jest/expect-expect
  it('Si tiene el placeholder esperado', () => {
    const component = render(<SearchBar />);
    const texto = 'Nunca dejes de buscar';
    component.getByPlaceholderText(texto);
  });
  it('Si hace match con el snapshot', () => {
    const { container } = render(<SearchBar />);
    expect(container).toMatchSnapshot();
  });
});
