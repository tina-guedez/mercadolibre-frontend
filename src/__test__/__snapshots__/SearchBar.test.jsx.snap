// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente SearchBar Si hace match con el snapshot 1`] = `
<div>
  <form
    action="/items"
    class="nav-search"
    method="GET"
    role="search"
  >
    <input
      aria-label="Ingresa el producto a buscar"
      autocomplete="off"
      class="nav-search-input"
      maxlength="120"
      name="q"
      placeholder="Nunca dejes de buscar"
      required=""
      tabindex="0"
      type="text"
    />
    <button
      class="nav-search-btn"
      tabindex="-1"
      type="submit"
    >
      <img
        aria-label="Buscar"
        class="nav-icon-search"
        src="test-file-stub"
      />
    </button>
  </form>
</div>
`;
